### ISSUE TITLE
aria-labelledby doesn't work with multiple references in IE 11

### ISSUE THREAD

**Issue by aminimalanimal** [nvaccess/nvda#5654]
NVDA 2015.4 in IE 11 does not correctly parse and read `aria-labelledby` when multiple `id` references are used.

This works properly in Firefox.

[Codepen - Dialogs with multiple `aria-labelledby` references](http://s.codepen.io/aminimalanimal/debug/VepKQN)

**Comment by aminimalanimal** [comment]
`aria-describedby` also chokes on multiple references within IE11.

**Comment by segedyjr** [comment]
It does work, but only if the references are to <label> elements.

**Comment by orvalle** [comment]
IE11 WORKAROUND: As with many issues getting IE11 to announce things adding tabindex="-1" to the wrapping tags used in aria-describedby will announce them.

Sample code (where two errors and a formatting instruction are included):

`
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<HTML lang="en" dir="ltr" xml:lang="en" xmlns:v="urn:schemas-microsoft-com:vml" xmlns="http://www.w3.org/1999/xhtml">
<HEAD>
<TITLE>Security Question</TITLE>				 
<BODY>
<FORM id="myForm" action="#">
	<DIV><LABEL for="userAnswer">What was mother's maiden name?</LABEL>
		<DIV id="challengeSecurityAnswerId">
			<INPUT name="userAnswer" class="challengeSecurityUserAnswerInput" id="userAnswer" aria-invalid="true" aria-required="true" aria-describedby="ErrorMsg1 ErrorMsg2 instruction1" type="password" size="50">
	
			<div id="instruction1" tabindex="-1">Answer is not case sensitive.</div>
			<div id="ErrorMsg1" tabindex="-1">Error 1: No answer was entered.</div>
			<div id="ErrorMsg2" tabindex="-1">Error 2: Your answer is incorrect.</div>
		</DIV>
	</DIV>
	
	<DIV>
		<INPUT name="_eventId" type="hidden" value="submit"> 
		<INPUT name="submit" type="submit" value="Next">
		<A href="#">Cancel</A>
	</DIV>
</FORM>
</BODY>
</HTML>
`

**Comment by andrew-l-d** [comment]
Even with only one use of aria-labelledby on the page, NVDA does not present the target in IE when not in forms mode without the tabindex="-1" hack on the target.  Other screen readers do not need the tabindex="-1".

Andrew

**Comment by feerrenrut** [comment]
@jcsteh Is this something we should address in the ARIA project?

**Comment by jcsteh** [comment]
[**@andrew-l-d**](https://github.com/andrew-l-d) commented on [31 Jan. 2017, 1:27 pm AEST](https://github.com/nvaccess/nvda/issues/5654#issuecomment-276265301 "2017-01-31T03:27:29Z - Replied by Github Reply Comments"):
> Even with only one use of aria-labelledby on the page, NVDA does not present the target in IE when not in forms mode without the tabindex="-1" hack on the target.  Other screen readers do not need the tabindex="-1".

@andrew-l-d: Can you please open a separate issue for this and include a test case? Thanks.

**Comment by Adriani90** [comment]
@aminimalanimal, @segedyjr, @orvalle can you still reproduce this issue? The codpen example seems to be expired and I cannot reproduce it. Could you upload a new example please? Thanks.

**Comment by aminimalanimal** [comment]
Sorry for the late response. It looks like Codepen doesn't keep “debug” mode active anymore, so the only thing that can be shared is a version with an iframe, which throws accessibility off.

I threw together a new example, but have yet to test it in NVDA as I don't have a virtual machine set up right now.
https://codepen.io/aminimalanimal/pen/bZNJxx

I've been really busy recently. Hopefully I'll find some time to test it, but I figured I'd drop it here in case someone else is able to before I am.
You might have to log in, fork the pen and click Debug mode for reliable results.

**Comment by aminimalanimal** [comment]
NVDA 2018.4 with Edge doesn't read any of the `aria-labelledby` or `aria-describedby`-linked content when the user enters a dialog. This is true even if I make it only a single reference in each attribute.

**Comment by Adriani90** [comment]
IE support has ended completely in 2022 and the browser is being removed automatically from machines that still have it after an Edge update in February 2023. I am closing this issue as abandoned, since I don't expect any further accessibility updates to IE or to IE mode in Edge. In Edge Chromium, Firefox or Chrome 111 this issue does not occur.
If you are having this issue in other browsers, please open a new issue with updated steps to reproduce and a minimal test case.
